### PR TITLE
draft: finished task2

### PR DESCRIPTION
**Pull request by YahorDauhaliuk**

### CODE REVIEW

File: models/DirWatcher.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		this.watchedFiles = [];
+		this.eventEmitter = eventEmitter;
+		this.watch(path, delay);
+	}
+
+	watch(dirPath, delay) {
+		const watchInterval = (callback) => setInterval(callback, delay);
+		const checkFolder = () => fs.readdir(dirPath, (err, watchingFiles) => {
+				if (err) {
+					throw err;
```
Comment (abruzgin) [inline models/DirWatcher.js]: better `throw new Error(err)`

File: models/Importer.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import csv from 'csvtojson';
```
Comment (abruzgin) [inline models/Importer.js]: Write your own CSV->JSON converter

File: models/Importer.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export default class Importer {
+	constructor (eventEmitter) {
+		this.eventEmitter = eventEmitter;
+		eventEmitter.on('dirWathcer:changed', (filePath) => this.import(filePath));
+	}
+
+	import (path) {
+		return new Promise((resolve) => {
+			csv()
+			.fromFile(path)
```
Comment (abruzgin) [inline models/Importer.js]: If you're using the chaining like this, please, make one indentation after the first method call, e.g.

meth()
--.child1()
--.child2()

**Review by abruzgin — APPROVED** [review]

File: models/DirWatcher.js (hunk 1 above)
Comment (YahorDauhaliuk) [reply models/DirWatcher.js]: fixed

File: models/Importer.js (hunk 2 above)
Comment (YahorDauhaliuk) [reply models/Importer.js]: Done.

File: models/DirWatcher.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+					const isModified = matchedFile && matchedFile.lastModified !== lastModified;
+
+					if (!matchedFile || isModified) {
+						const newFile = { name: fileName, lastModified };
+						if (isModified) {
+							this.watchedFiles[matchedFileIndex] = newFile;
+						} else {
+							this.watchedFiles.push(newFile);
+						}
+						const filePath = `${dirPath}/${fileName}`;
```
Comment (abruzgin) [inline models/DirWatcher.js]: Move it to the upper scope to use in `lastModified`

File: package.json [hunk 5]
```diff
@@ -15,9 +15,14 @@
     "babel-preset-env": "^1.6.1",
     "babel-preset-stage-2": "^6.24.1",
     "babel-register": "^6.26.0",
+    "eventemitter2": "^5.0.1",
```
Comment (abruzgin) [inline package.json]: Please, use built-in `events` NodeJS package for EventEmitter

Comment (YahorDauhaliuk) [reply package.json]: Done

File: models/DirWatcher.js (hunk 4 above)
Comment (YahorDauhaliuk) [reply models/DirWatcher.js]: Done

File: models/DirWatcher.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+				}
+
+				watchingFiles.forEach((fileName) => {
+					if (fileName.indexOf('csv') === -1) {
+						return;
+					}
+
+					const lastModified = fs.statSync(`${dirPath}/${fileName}`).mtime.getTime();
+					const matchedFile = this.watchedFiles.find(({ name }) => name === fileName);
+					const filePath = `${dirPath}/${fileName}`;
```
Comment (abruzgin) [inline models/DirWatcher.js]: Move it 2 lines upper as you use it in `lastModified` =)

**Pull request merged**